src/modules/hip/kernel/image_sum.hpp

```diff
@@ -0,0 +1,481 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+#include "reduction.hpp"
+
+// -------------------- Set 0 - Reduction Stage 2 --------------------
+
+__global__ void image_sum_grid_result_tensor(float *srcPtr,
+                                             uint xBufferLength,
+                                             float *dstPtr)
+{
+    int id_x = hipThreadIdx_x * 8;
+    int id_z = hipBlockIdx_z;
+
+    __shared__ float partialSumLDS[1024];                            // 2048 floats of src reduced to 1024 in a 1024 x 1 thread block
+    partialSumLDS[hipThreadIdx_x] = 0.0f;                           // initialization of LDS to 0 using all 1024 x 1 threads
+
+    if (id_x >= xBufferLength)
+    {
+        return;
+    }
+
+    int xAlignedLength = xBufferLength & ~7;                        // alignedLength for vectorized global loads
+    int xDiff = xBufferLength - xAlignedLength;                     // difference between bufferLength and alignedLength
+    uint srcIdx = (id_z * xBufferLength) + id_x;
+
+    d_float8 src_f8;
+    rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &src_f8);   // load 8 pixels to local mmemory
+    if (id_x + 8 > xBufferLength)
+        for(int i = xDiff; i < 8; i++)
+            src_f8.f1[i] = 0.0f;                                    // local memory reset of invalid values (from the vectorized global load) to 0.0f
+    src_f8.f4[0] += src_f8.f4[1];                                   // perform small work of vectorized float4 addition
+    partialSumLDS[hipThreadIdx_x] += (src_f8.f1[0] +
+                                      src_f8.f1[1] +
+                                      src_f8.f1[2] +
+                                      src_f8.f1[3]);                // perform small work of reducing float4s to float using 256 x 1 threads and store in LDS
+    __syncthreads();                                                // syncthreads after LDS load
+
+    // Reduction of 1024 floats on 1024 threads per block in x dimension
+    for (int threadMax = 512; threadMax >= 1; threadMax /= 2)
+    {
+        if (hipThreadIdx_x < threadMax)
+            partialSumLDS[hipThreadIdx_x] += partialSumLDS[hipThreadIdx_x + threadMax];
+        __syncthreads();
+    }
+
+    // Final store to dst
+    if (hipThreadIdx_x == 0)
+        dstPtr[hipBlockIdx_z] = partialSumLDS[0];
+}
+
+__global__ void image_sum_grid_3channel_result_tensor(float *srcPtr,
+                                                      uint xBufferLength,
+                                                      float *dstPtr)
+{
+    int id_x = hipThreadIdx_x * 8;
+    int id_z = hipBlockIdx_z;
+
+    __shared__ float partialRSumLDS[1024];                            // 2048 floats of src reduced to 1024 in a 1024 x 1 thread block
+    __shared__ float partialGSumLDS[1024];
```

Comment:
Please check how many floats we can handle in each lds allocation. For 1024 threads, each loading 8, it might be 1024 * 8 = 8192, so please check on accuracy of all comments too